pydantic/_internal/_model_construction.py

```diff
@@ -553,6 +556,59 @@ def complete_model_class(
     return True
 
 
+def set_deprecated_descriptors(cls: type[BaseModel]) -> None:
+    """Set data descriptors on the class for deprecated fields."""
+    for field, field_info in cls.model_fields.items():
+        if (msg := field_info.deprecation_message) is not None:
+            desc = _DeprecatedFieldDescriptor(msg)
+            desc.__set_name__(cls, field)
+            setattr(cls, field, desc)
+
+    for field, computed_field_info in cls.model_computed_fields.items():
+        if (
+            not computed_field_info.from_deprecated_decorator  # Avoid having two warnings emitted
```

Comment:
Similar to my other comments, if we found a way around having this set on each computed field info instance, we could streamline the logic for model fields and computed fields.

It's ok if we can't do that, but just wanted to note here, as it's relevant in this func.